doc/model.md

```diff
@@ -0,0 +1,74 @@
+# Istio Service Model
+
+Istio's services model introduces the concept of a service version, a
+finer-grained service notion used to subdivide service instances by
+versions (`v1`, `v2`) or environment (`staging`, `prod`). Istio routing
+rules can refer to the service versions, to provide additional control over
+traffic between services.
+
+The Istio model of services in the application is independent of the
+underlying platform (Kubernetes, Mesos, CloudFoundry, etc.). Platform
+specific adapters are responsible for populating the internal model
+representation with various fields, from the metadata found in the
+platform. The following sections describe various aspects of the abstract
+service model used by Istio.
+
+## Service
+
+Service is a unit of an application with a unique name that other services
```

Comment:
Suggestions:

Modern cloud applications are created by composing together independent sets of services. Individual services are referenced by their fully-qualified domain name and one or more ports where the service is listening for connections.